SeniorInterview/StocksAnalyzed/src/AggregateRequestor.java

```diff
@@ -0,0 +1,59 @@
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.ArrayList;
+
+public class AggregateRequestor extends Requestor {
+
+    public String request(String ticker, String startDate, String endDate){
+        String grouppedURL = String.format("https://api.polygon.io/v2/aggs/ticker/%s/range/1/day/%s/%s?adjusted=true&sort=asc&limit=120&apiKey=%s",
+                ticker,
+                startDate,
+                endDate,
+                apiKey);
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(grouppedURL))
+                .method("GET", HttpRequest.BodyPublishers.noBody())
+                .build();
+        HttpResponse<String> response = null;
+        try{
+            response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString());
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        return response.body();
+    }
+
+    public Ticker requestTickerDays(String tickerName, String startDate, String endDate){
+        String response = request(tickerName, startDate, endDate);
+        JSONObject jsonResponse = responseToJSON(response);
+        JSONArray results = (JSONArray) jsonResponse.get("results");
+
+        Ticker ticker = null;
+
+        for (Object it : results){
+            JSONObject tickerJson = (JSONObject) it;
+            double open = ToDouble(tickerJson.get("o"));
```

Comment:
same as comment above
always good to have String as consts in this  kind of context, makes easier to update

Reply:
for this one you could also have an Enum with the name and letter it represents (ex: OPEN = "o"), this way you can use same here and on TickerDay, for instance